internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -318,7 +318,15 @@ func resourceMonitorDiagnosticSettingRead(d *pluginsdk.ResourceData, meta interf
 	}
 
 	d.Set("name", id.Name)
-	d.Set("target_resource_id", id.ResourceUri)
+
+	resourceId, err := azure.ParseAzureResourceID(actualResourceId)
```

Comment:
this won't work if this item is Subscription-scoped?

Reply:
Yes, current method only copes with `/resourcegroup/` -> `/resourceGroup/`. For `/subscription/`, it is always all lower cases. But for other segments like `/microsoft.network/` -> `/Microsoft.Network/` or `/storageaccounts/` -> `/storageAccounts/`, these cases need a more general method.